Frontend/ui-driver/src/Screens/RideRequestScreen/Controller.purs

```diff
@@ -0,0 +1,344 @@
+import Prim.Boolean (False)
+import Screens (ScreenName(..), getScreen)
+import Data.Function.Uncurried (runFn1)
+import Engineering.Helpers.Commons as EHC
+import JBridge as JBridge
+import Debug
+import PrestoDOM.Types.Core (class Loggable, toPropValue)
+import Screens.Types (AnimationState(..), RideRequestCardState, ItemState, RideCardItemState)
+import Components.RideRequestCard.Controller as RideRequestCardActionController
+import Services.API
+import Engineering.Helpers.Commons (convertUTCtoISC)
+import Data.Maybe
+import Services.API
+import Common.Types.App as CTA
+import Resource.Constants
+import JBridge (fromMetersToKm, differenceBetweenTwoUTC)
+import Data.Array (mapWithIndex, (!!), length, null)
+import Data.Function.Uncurried (runFn2)
+import Helpers.Utils as HU
+import Engineering.Helpers.Commons
+import Data.Maybe (Maybe(..), fromMaybe)
+import Data.String (Pattern(..), split, length, take, drop, joinWith, trim)
+import Data.Int (fromString)
+import Styles.Colors as Color
+import Data.Int as Int
+import Prelude (class Eq, class Show, Unit, identity, unit, ($), (/=), (==), (<>), (<<<))
+import Screens.Types as ST
+import Data.Array as DA
+
+
+
+instance showAction :: Show Action where
+  show _ = ""
+
+instance loggableAction :: Loggable Action where
+  performLog = defaultPerformLog
+
+data ScreenOutput
+  = GoBack
+  -- | UpdateRoute
+  -- | GoToHelp
+  | LoaderOutput RideRequestScreenState
+  | RefreshScreen RideRequestScreenState
+  | GoToRideSummary RideRequestScreenState
+  | FcmNotifications String RideRequestScreenState
+
+data Action
+  = BackPressed
+  | UpdateOverLayHeight
+  | ExpireOverLayAfterRender
+  | SelectDay Int
+  | OnCardclick
+  | DropDown
+  | NoAction
+  | RideRequestCardActionController RideRequestCardActionController.Action
+  | PastRideApiAC ScheduledBookingListResponse String
+  | GetRideList ScheduledBookingListResponse String
+  | Refresh
+  | ShowMap String String String
+  | AfterRender
+  | RideTypeSelected (Maybe (CTA.TripCategoryTag)) Int
+  | FilterSelected
+  | OnClickToime Int
+  | Scroll String
+  | ScrollStateChanged ScrollState
+  | Loader
+  | OnFadeComplete String
+  | Notification String
+
+
+eval :: Action -> RideRequestScreenState -> Eval Action ScreenOutput RideRequestScreenState
+eval BackPressed state = exit GoBack
+
+eval (SelectDay index) state = do
+  let
+    updatedItem = map (\item -> if item.index == index then item { isSelected = true } else item { isSelected = false }) state.data.dayArray
+
+    newState = state { data { dayArray = updatedItem , activeDayIndex = index , shimmerLoader  = ST.AnimatedIn} }
+  continueWithCmd newState [pure $ FilterSelected]
+
+eval OnCardclick state = continue state
+
+eval UpdateOverLayHeight state = continue state { props { cardHeight = (runFn1 JBridge.getLayoutBounds $ EHC.getNewIDWithTag "cardddd").height } }
+
+eval ExpireOverLayAfterRender state = continue state
+
+eval DropDown state = continue state
```

Comment:
Why so many actions have continue state. We can have one NoAction and use it everywhere.